### ISSUE TITLE
Network does not converge without prei-training

### ISSUE THREAD

**Issue by GeoZcx** [wgcban/ChangeFormer#17]
Hi, Chaminda, Thanks for your code! Nice work! I trained the ChangeFormer_v6 without loading the pre-trained files and the network did not converge on any dataset. It seems that the only choice is to load the pre-trained file, otherwise, it would not converge with very large losses. I notice that once using the pre-trained file, the network can perform well as expected. So, have you ever tried training ChangeFormer_v6 without pre-training? And what's the result? Did it converge very slowly or not converge? Thanks!

**Comment by wgcban** [comment]
Hi @GeoZcx 

Thank you for your interest in our work!

I tried to train the ChangeFormer without loading the pre-trained trained weights in the initial phase of the project and the network converged. However, in this case, the network converged a bit slow compared to the model initialized with pre-trained weights, so I always used pre-trained weights for my other experiments to save time. Even when I was training ChangeFormer for the DSIFN dataset, I used the pre-trained weights in LEVIRD-CD to get much faster convergence. Since you were seeing a very high loss value when you weren't using pre-trained weights, I recommend that you reduce the learning later by a factor of 10 or 100 so that the network slowly converges. Especially, if you are using the AdamW optimizer, I highly recommend that you use a very low learning rate compared to using Adam or SGD because AdamW has a very large boost term, so it requires a much lower learning rate. Also, using the pre-trained weight is very common when training transformer networks (i.e., non-local attention networks), so I recommend following the same trend in your research as well.

Let me know if you have further questions.

Best,
Chaminda.

**Comment by wgcban** [comment]
Hi @GeoZcx 

Also, one more thing I forgot to highlight in my previous answer. I have experienced improved performance when I was training my ChangeFormer with ``multi_scale-loss``. Please make sure you have turned on this feature in the script when you are training the model.

Best,

Chaminda.
